code/modules/mob/mob_movement.dm

```diff
@@ -302,7 +302,7 @@
 
 		//We are now going to move
 		move_delay = max(move_delay,1)
-		if(mob.movement_speed_modifier)
+		if(mob.movement_speed_modifier && mob.movement_speed_modifier != 1)
```

Comment:
I didn't do it because of runtimes.
There's just no reason to run that code if the modifier is 1, since it would just multiply the speed by 1/1, which would do nothing.